cypress/integration/unit_tests/checkbox_test.js

```diff
@@ -176,6 +178,103 @@ describe('Unit test for toggleLayerOn', () => {
     });
   });
 
+  it('fails to load a layer', () => {
+    setUpLayerFailure().then(
+        () => addLayer(
+            {
+              'ee-name': 'asset/does/not/exist',
+              'asset-type': LayerType.FEATURE_COLLECTION,
+              'display-name': 'asset1',
+              'display-on-load': false,
+              'color-function': colorProperties,
+              'index': 3,
+            },
+            null));
+    assertLayerFailure();
+  });
+
+  it('fails to toggle on a layer', () => {
+    const layer = {
+      'ee-name': 'asset/does/not/exist',
+      'asset-type': LayerType.FEATURE_COLLECTION,
+      'display-name': 'asset1',
+      'display-on-load': false,
+      'color-function': colorProperties,
+      'index': failureLayerIndex,
+    };
+    addNullLayer(layer);
+    setUpLayerFailure().then(
+        () => toggleLayerOn(
+            {
+              'ee-name': 'asset/does/not/exist',
+              'asset-type': LayerType.FEATURE_COLLECTION,
+              'display-name': 'asset1',
+              'display-on-load': false,
+              'color-function': colorProperties,
+              'index': failureLayerIndex,
+            },
+            null));
+    assertLayerFailure();
+  });
+
+  it('fails to load an image layer', () => {
+    setUpLayerFailure().then(
+        () => addLayer(
+            {
+              'ee-name': 'asset/does/not/exist',
+              'asset-type': LayerType.IMAGE,
+              'display-name': 'asset1',
+              'display-on-load': false,
+              'color-function': colorProperties,
+              'index': failureLayerIndex,
+            },
+            null));
+    assertLayerFailure();
+  });
+
+  const failureLayerIndex = 3;
+
+  /**
+   * Prepares for a test that fails to load a layer.
+   * @return {Cypress.Chainable<Sinon.SinonStub>}
+   */
+  function setUpLayerFailure() {
+    const index = 3;
+    // Visit a blank page first to clear out any prior page state.
+    cy.visit('test_utils/empty.html');
+    cy.document().then((doc) => {
+      const div = doc.createElement('div');
+      div.id = getCheckBoxRowId(index);
+      const checkbox = doc.createElement('input');
+      checkbox.id = getCheckBoxId(index);
+      div.appendChild(checkbox);
+      doc.body.appendChild(div);
+      cy.stub(document, 'getElementById')
+          .callsFake((id) => doc.getElementById(id));
+    });
+    return cy.wrap(
```

Comment:
difference between this strategy and using a stub constant that's accessible?

Reply:
This doesn't pollute global namespace. Just a matter of preference. Slight advantage to this approach is you get Cypress execution order for "free", because of the `cy.get('@...'` on the other end.